### PR TITLE
refactor: remove unused deps and simplify

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This does a few things:
- Removes a bunch of un-used dev packages
- Prefers `Object.assign`
- Untangles the incomprehensible mess that was the usage of `is`

### CODE REVIEW

File: src/interfaces/express.ts [hunk 1]
```diff
@@ -65,15 +62,13 @@ export function makeExpressHandler(
 
     populateErrorMessage(err, em);
 
-    if (isObject(client) && isFunction(client.sendError)) {
-      client.sendError(em);
-    }
+    if (is.object(client) && is.function(client.sendError))
+      { client.sendError(em); }
```
Comment (kinwa91) [inline src/interfaces/express.ts]: Syntax here looks a bit messed up

File: src/interfaces/hapi.ts [hunk 2]
```diff
@@ -101,18 +100,20 @@ export function makeHapiPlugin(client: RequestHandler, config: Configuration) {
               }
             });
       } else {
-        if (isFunction(server.on)) {
-          server.on('request-error', (req: hapi.Request, err: {}) => {
-            client.sendError(hapiErrorHandler(err, req, config));
-          });
-        }
+        if (is.function(server.on))
```
Comment (kinwa91) [inline src/interfaces/hapi.ts]: weird syntax

File: src/interfaces/express.ts (hunk 1 above)
Comment (JustinBeckwith) [reply src/interfaces/express.ts]: The syntax is fine, it's `clang-format` doing *weird* stuff. It still runs fine.

File: src/interfaces/hapi.ts (hunk 2 above)
Comment (JustinBeckwith) [reply src/interfaces/hapi.ts]: same as above :(

**Pull request merged**